Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -50,9 +51,10 @@ public final class JSONRPCConection {
 
   var closeHandler: () -> Void
 
-  public init(inFD: Int32, outFD: Int32, closeHandler: @escaping () -> Void = {}) {
+  public init(messageRegistry: MessageRegistry, inFD: Int32, outFD: Int32, closeHandler: @escaping () -> Void = {}) {
```

Comment:
Maybe we could call this field "protocol" since it's basically providing the application layer protocol?

Reply:
yes, that is better, will update

Reply:
well, except its a reserved keyword so would have to use funky backticks. `rpcProtocol`?

Reply:
`public init(protocol messageRegistry: MessageRegistry ...)` should work without backticks at the declaration or call site.  It's only treated as a keyword when used on its own.

Reply:
TIL